### PR TITLE
Backend code review

### PR DESCRIPTION
**Pull request by Ayush-kumar1**
Please review the backend code for social media app

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const app = express();
+const PORT = 5000;
+import User from "./models/user.js"
+import Post from "./models/post.js";
+import authRoutes from "./routes/auth.js";
+import postRouter from "./routes/post.js";
+import userRouter from "./routes/user.js";
+
+// password-PhFK3sbXy5WZJeG4
+const dbURL = "mongodb+srv://admin:PhFK3sbXy5WZJeG4@cluster0.nduys.mongodb.net/SocialMediaDB?retryWrites=true&w=majority";
```
Comment (rohiraanjali) [inline server.js:13]: Backend review -> starting. 

This is a secret key and should not be open to everyone. store it in .env/key.js file and then to .gitignore

File: server.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+app.use(express.json())
+app.use(cors());
+mongoose.connect(dbURL, {
```
Comment (rohiraanjali) [inline server.js:16]: Make a separate file and folder for the Database and store it there.

File: routes/auth.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    if (!email || !name || !password) {
+        return res.status(401).json({ error: "Please fill all fields completely" })
+    }
+
+    User.findOne({ email: email })
+        .then((saveduser) => {
+            if (saveduser) {
+                return res.status(422).json({ error: "user already exists" })
+            }
```
Comment (rohiraanjali) [inline routes/auth.js:29]: The code inside the routes file is huge. Make a controller folder and put all the CRUD operations there and just refer the functions name in routes files when you do get, post, delete, update. Do this for all the files of routes i.e. auth, user, post.

File: routes/auth.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    User.findOne({ email })
+        .then(savedUser => {
+            if (!savedUser) {
+                return res.json({ message: "This user does not exist" })
+            }
+
+            bcrypt.compare(password, savedUser.password)
+                .then(doMatch => {
+                    if (doMatch) {
+                        // return res.json({ message: "Login sucessful" })
```
Comment (rohiraanjali) [inline routes/auth.js:74]: remove all the commented-out code through out the codebase

**Review by rohiraanjali — CHANGES_REQUESTED** [review]
There are some changes in the backend code as well. Look at the PR comments and make changes accordingly.

File: key.js [hunk 5]
```diff
@@ -0,0 +1 @@
+export const JWT_SECRET = "ayush_Token";
```
Comment (thesudeshdas) [inline key.js:1]: This should be hidden

**Review by thesudeshdas — CHANGES_REQUESTED** [review]

**Pull request merged**